File: web/controllers/landing.py
from __future__ import annotations

from web.auth import current_user
from web.http_util import Request, Response, attach_csrf_cookie, html_response, redirect


def index(request: Request) -> Response:
    user = current_user(request)
    if user:
        return redirect("/sites")
    response = html_response(
        "landing/index.html",
        title="example-monitoring",
        csrf_token=request.csrf_token,
    )
    return attach_csrf_cookie(response, request.csrf_token)

File: web/src/controllers/landing.rs
use std::collections::HashMap;

use axum::extract::State;
use axum::http::HeaderMap;
use minijinja::Value;

use crate::auth;
use crate::http::{AppResponse, AppState, RequestContext};

pub async fn index(State(state): State<AppState>, headers: HeaderMap) -> AppResponse {
    let ctx = RequestContext::from_headers(&headers, &state.config.secret_key);
    if auth::current_user(&state.db, &ctx).is_some() {
        return AppResponse::redirect(&state, &ctx, "/sites");
    }
    let mut vars = HashMap::new();
    vars.insert("title".into(), Value::from("example-monitoring"));
    AppResponse::html(&state, &ctx, "landing/index.html", true, vars)
}
